packages/core/ui-settings/core-ui-settings-common/src/ui_settings.ts

```diff
@@ -57,6 +57,12 @@ export interface UiSettingsParams<T = unknown> {
   requiresPageReload?: boolean;
   /** a flag indicating that value cannot be changed */
   readonly?: boolean;
+  /**
+   * a flag indicating the level of restriction of the readonly settings
+   * 'strict' indicates that the value cannot be changed through API and is not displayed in the UI
+   * 'ui' indicates that the value is just not displayed in the UI
+   * */
+  readonlyMode?: 'strict' | 'ui';
```

Comment:
Currently, settings with `readonly: true` are [filtered out from the UI](https://github.com/elastic/kibana/blob/4b02740b32542b6e84ee7a8abec9cb355f70db35/src/plugins/advanced_settings/public/management_app/settings_helper.ts#L26), but there is no enforcement for prohibiting `set` on these settings. Since we want to prohibit `set` for non-allowlisted settings, the idea is to add two readonly modes - `ui`, which basically has the current behaviour of `readonly: true`, and `strict`, which also restricts `set`. We want to add these modes so that these changes don't affect the current behaviour of any of the existing settings that are readonly.

We started discussing this approach with @clintandrewhall and @lukeelmers last week (see https://github.com/elastic/kibana/pull/164471#discussion_r1304396905 and https://github.com/elastic/kibana/pull/164471#discussion_r1305667169) but we haven't clarified some details yet - we have a sync-up today so we will discuss this further.